src/components/radio/RadioGroup.tsx

```diff
@@ -50,9 +48,12 @@ class RadioGroup extends React.Component<RadioGroupProps, RadioGroupState> {
         return (
             <div className={`radio-group ${className}`} onChange={this.onChangeHandler}>
                 {React.Children.map(children, (radio: React.ReactElement) => {
-                    const { value, ...rest } = radio.props;
+                    const { value } = radio.props;
 
-                    return <RadioButton isSelected={value === stateValue} name={name} value={value} {...rest} />;
+                    return React.cloneElement(radio, {
+                        name,
+                        isSelected: value === stateValue,
```

Comment:
Would this miss other props ?

Reply:
`cloneElement` will preserve the original element's props. We only need to pass the ones that are set dynamically by `RadioGroup`, which will be shallow merged with the rest.